apps/backend/src/libs/modules/controller/base-controller.module.ts

```diff
@@ -34,6 +35,16 @@ class BaseController implements Controller {
 		return await reply.status(status).send(payload);
 	}
 
+	private mapPreHandler({
```

Comment:
We don't need mapPreHandler. We should add preHandlers directly to routeParameters in `base-server-application.ts` addRoute method. The preHandler function will receive request, reply, done and proceed with the logic

Reply:
1. In user controller I created pre handler function with type `(options: APIHandlerOptions) => void`.
2. In base controller I use mapPreHandler to convert preHandlers from `(options: APIHandlerOptions) => void` to `(request: FastifyRequest, reply: FastifyReply, done: HookHandlerDoneFunction,) => void`.
3. In base server application in addRoute I transfered all preHandlers to app.route.
Tried to do everything as with handles.
I don't understand, should I create function with `(request: FastifyRequest, reply: FastifyReply, done: HookHandlerDoneFunction,) => void` type right in user controller?

Reply:
Yes, you should create function with `(request: FastifyRequest, reply: FastifyReply, done: HookHandlerDoneFunction,) => void` type. You should create it in a separate file though